--- php_mnogo.c	2012-05-06 12:30:50.087426746 +0200
+++ php_mnogo.c.oden	2012-05-06 12:31:41.776783531 +0200
@@ -151,7 +151,7 @@ static int le_link,le_res;
 
 /* {{{ mnogosearch_functions[]
  */
-function_entry mnogosearch_functions[] = {
+zend_function_entry mnogosearch_functions[] = {
 	PHP_FE(udm_api_version,		NULL)
 
 #if UDM_VERSION_ID >= 30200			
@@ -470,7 +470,7 @@ DLEXPORT PHP_FUNCTION(udm_alloc_agent)
 	switch(ZEND_NUM_ARGS()){
 
 		case 1: {
-				pval **yydbaddr;
+				zval **yydbaddr;
 				char *dbaddr;
 				UDM_ENV   * Env;
 				UDM_AGENT * Agent;
@@ -527,8 +527,8 @@ DLEXPORT PHP_FUNCTION(udm_alloc_agent)
 			break;
 			
 		case 2: {
-				pval **yydbaddr;
-				pval **yydbmode;
+				zval **yydbaddr;
+				zval **yydbmode;
 				char *dbaddr;
 				char *dbmode;
 				UDM_ENV   * Env;
@@ -597,7 +597,7 @@ DLEXPORT PHP_FUNCTION(udm_alloc_agent)
    Set mnoGoSearch agent session parameters */
 DLEXPORT PHP_FUNCTION(udm_set_agent_param)
 {
-	pval **yyagent, **yyvar, **yyval;
+	zval **yyagent, **yyvar, **yyval;
 	char *val;
 	int var;
 	UDM_AGENT * Agent;
@@ -1140,7 +1140,7 @@ DLEXPORT PHP_FUNCTION(udm_set_agent_para
    Load ispell data */
 DLEXPORT PHP_FUNCTION(udm_load_ispell_data)
 {
-	pval **yyagent, **yyvar, **yyval1, **yyval2, **yyflag, **yycharset ;
+	zval **yyagent, **yyvar, **yyval1, **yyval2, **yyflag, **yycharset ;
 	char *val1, *val2, *charset;
 	int var, flag;
 	UDM_AGENT * Agent;
@@ -1302,7 +1302,7 @@ DLEXPORT PHP_FUNCTION(udm_load_ispell_da
    Free memory allocated for ispell data */
 DLEXPORT PHP_FUNCTION(udm_free_ispell_data)
 {
-	pval ** yyagent;
+	zval ** yyagent;
 	UDM_AGENT * Agent;
 	switch(ZEND_NUM_ARGS()){
 		case 1: {
@@ -1335,7 +1335,7 @@ DLEXPORT PHP_FUNCTION(udm_free_ispell_da
    Add mnoGoSearch search restrictions */
 DLEXPORT PHP_FUNCTION(udm_add_search_limit)
 {
-	pval **yyagent, **yyvar, **yyval;
+	zval **yyagent, **yyvar, **yyval;
 	char *val;
 	int var;
 	UDM_AGENT * Agent;
@@ -1476,7 +1476,7 @@ DLEXPORT PHP_FUNCTION(udm_add_search_lim
    Clear all mnoGoSearch search restrictions */
 DLEXPORT PHP_FUNCTION(udm_clear_search_limits)
 {
-	pval ** yyagent;
+	zval ** yyagent;
 	UDM_AGENT * Agent;
 	int i;
 	
@@ -1516,7 +1516,7 @@ DLEXPORT PHP_FUNCTION(udm_clear_search_l
    Check if the given charset is known to mnogosearch */
 DLEXPORT PHP_FUNCTION(udm_check_charset)
 {
-	pval ** yycharset, ** yyagent;
+	zval ** yycharset, ** yyagent;
 	UDM_AGENT * Agent;
 	int id=-1;
 
@@ -1549,7 +1549,7 @@ DLEXPORT PHP_FUNCTION(udm_check_charset)
    Return CRC32 checksum of gived string */
 DLEXPORT PHP_FUNCTION(udm_crc32)
 {
-	pval ** yystr, ** yyagent;
+	zval ** yystr, ** yyagent;
 	char *str;
 	int crc32;
 	char buf[32];
@@ -1584,7 +1584,7 @@ DLEXPORT PHP_FUNCTION(udm_crc32)
    Parses query string, initialises variables and search limits taken from it */
 DLEXPORT PHP_FUNCTION(udm_parse_query_string)
 {
-	pval ** yystr, ** yyagent;
+	zval ** yystr, ** yyagent;
 	char *str;
 	UDM_AGENT * Agent;
 	int id=-1;
@@ -1613,7 +1613,7 @@ DLEXPORT PHP_FUNCTION(udm_parse_query_st
    Perform search */
 DLEXPORT PHP_FUNCTION(udm_make_excerpt)
 {
-	pval ** yyagent, **yyres, **yyrow_num;
+	zval ** yyagent, **yyres, **yyrow_num;
 	UDM_RESULT * Res;
 	UDM_AGENT * Agent;
 	int id=-1, row;
@@ -1683,7 +1683,7 @@ DLEXPORT PHP_FUNCTION(udm_make_excerpt)
    Set mnoGoSearch agent session parameters extended */
 DLEXPORT PHP_FUNCTION(udm_set_agent_param_ex)
 {
-	pval **yyagent, **yyvar, **yyval;
+	zval **yyagent, **yyvar, **yyval;
 	char *val, *var;
 	UDM_AGENT * Agent;
 
@@ -1741,7 +1741,7 @@ DLEXPORT PHP_FUNCTION(udm_set_agent_para
    Fetch mnoGoSearch environment parameters */
 DLEXPORT PHP_FUNCTION(udm_get_agent_param_ex)
 {
-	pval **yyagent, **yyfield_name;
+	zval **yyagent, **yyfield_name;
 
 	UDM_AGENT * Agent;
 	int row;
@@ -1771,7 +1771,7 @@ DLEXPORT PHP_FUNCTION(udm_get_agent_para
    Fetch mnoGoSearch result field */
 DLEXPORT PHP_FUNCTION(udm_get_res_field_ex)
 {
-	pval **yyres, **yyrow_num, **yyfield_name;
+	zval **yyres, **yyrow_num, **yyfield_name;
 
 	UDM_RESULT * Res;
 	int row;
@@ -1815,7 +1815,7 @@ DLEXPORT PHP_FUNCTION(udm_get_res_field_
    Get CachedCopy of document and return TRUE if cached copy found */
 DLEXPORT PHP_FUNCTION(udm_store_doc_cgi)
 {
-	pval ** yyagent;
+	zval ** yyagent;
 	UDM_AGENT * Agent;
 	int id=-1;
 	
@@ -1922,7 +1922,7 @@ DLEXPORT PHP_FUNCTION(udm_alloc_agent_ar
 	switch(ZEND_NUM_ARGS()){
 
 		case 1: {
-				pval **yydbaddr;
+				zval **yydbaddr;
 				zval **tmp;
 				char *dbaddr;
 				UDM_ENV   * Env;
@@ -1974,7 +1974,7 @@ DLEXPORT PHP_FUNCTION(udm_alloc_agent_ar
    Return Hash32 checksum of gived string */
 DLEXPORT PHP_FUNCTION(udm_hash32)
 {
-	pval ** yystr, ** yyagent;
+	zval ** yystr, ** yyagent;
 	char *str;
 	udmhash32_t hash32;
 	char buf[32];
@@ -2012,7 +2012,7 @@ DLEXPORT PHP_FUNCTION(udm_hash32)
    Perform search */
 DLEXPORT PHP_FUNCTION(udm_find)
 {
-	pval ** yyquery=NULL, ** yyagent;
+	zval ** yyquery=NULL, ** yyagent;
 	UDM_RESULT * Res;
 	UDM_AGENT * Agent;
 	int id=-1;
@@ -2062,7 +2062,7 @@ DLEXPORT PHP_FUNCTION(udm_find)
    Fetch mnoGoSearch result field */
 DLEXPORT PHP_FUNCTION(udm_get_res_field)
 {
-	pval **yyres, **yyrow_num, **yyfield_name;
+	zval **yyres, **yyrow_num, **yyfield_name;
 
 	UDM_RESULT * Res;
 	int row,field;
@@ -2247,7 +2247,7 @@ DLEXPORT PHP_FUNCTION(udm_get_res_field)
    Get mnoGoSearch result parameters */
 DLEXPORT PHP_FUNCTION(udm_get_res_param)
 {
-	pval ** yyres, ** yyparam;
+	zval ** yyres, ** yyparam;
 	int param;
 	UDM_RESULT * Res;
 	switch(ZEND_NUM_ARGS()){
@@ -2360,7 +2360,7 @@ DLEXPORT PHP_FUNCTION(udm_get_res_param)
    mnoGoSearch free result */
 DLEXPORT PHP_FUNCTION(udm_free_res)
 {
-	pval ** yyres;
+	zval ** yyres;
 	UDM_RESULT * Res;
 	switch(ZEND_NUM_ARGS()){
 		case 1: {
@@ -2384,7 +2384,7 @@ DLEXPORT PHP_FUNCTION(udm_free_res)
    Free mnoGoSearch session */
 DLEXPORT PHP_FUNCTION(udm_free_agent)
 {
-	pval ** yyagent;
+	zval ** yyagent;
 	UDM_RESULT * Agent;
 	switch(ZEND_NUM_ARGS()){
 		case 1: {
@@ -2408,7 +2408,7 @@ DLEXPORT PHP_FUNCTION(udm_free_agent)
    Get mnoGoSearch error number */
 DLEXPORT PHP_FUNCTION(udm_errno)
 {
-	pval ** yyagent;
+	zval ** yyagent;
 	UDM_AGENT * Agent;
 	switch(ZEND_NUM_ARGS()){
 		case 1: {
@@ -2440,7 +2440,7 @@ DLEXPORT PHP_FUNCTION(udm_errno)
    Get mnoGoSearch error message */
 DLEXPORT PHP_FUNCTION(udm_error)
 {
-	pval ** yyagent;
+	zval ** yyagent;
 	UDM_AGENT * Agent;
 	
 	switch(ZEND_NUM_ARGS()){
@@ -2475,7 +2475,7 @@ DLEXPORT PHP_FUNCTION(udm_api_version)
    Get mnoGoSearch categories list with the same root */
 DLEXPORT PHP_FUNCTION(udm_cat_list)
 {
-	pval ** yycat, ** yyagent;
+	zval ** yycat, ** yyagent;
 	UDM_AGENT * Agent;
 	char *cat;
 #if UDM_VERSION_ID >= 30204	
@@ -2549,7 +2549,7 @@ DLEXPORT PHP_FUNCTION(udm_cat_list)
    Get mnoGoSearch categories path from the root to the given catgory */
 DLEXPORT PHP_FUNCTION(udm_cat_path)
 {
-	pval ** yycat, ** yyagent;
+	zval ** yycat, ** yyagent;
 	UDM_AGENT * Agent;
 	char *cat;
 #if UDM_VERSION_ID >= 30204	
@@ -2624,7 +2624,7 @@ DLEXPORT PHP_FUNCTION(udm_cat_path)
    Get total number of documents in database */
 DLEXPORT PHP_FUNCTION(udm_get_doc_count)
 {
-	pval ** yyagent;
+	zval ** yyagent;
 	UDM_AGENT * Agent;
 	int id=-1;
 
